src/error.rs

```diff
@@ -50,7 +50,7 @@ pub type Result<T> = std::result::Result<T, Error>;
 /// [`ErrorKind`](enum.ErrorKind.html) is wrapped in an `Arc` to allow the errors to be
 /// cloned.
 #[derive(Clone, Debug, Error)]
-#[error("Kind: {kind}, labels: {labels:?}")]
+#[error("Kind: {kind}, labels: {labels:?}, backtrace: {bt}")]
```

Comment:
Having a backtrace from the point of error _construction_ rather than the point of handling was very useful for debugging a lot of failures here, and it doesn't cost anything to construct unless `RUST_BACKTRACE` is set.  This does make error messages quite a bit more verbose; open to thoughts on how this should work.

Reply:
This seems valuable for us, but maybe put it behind `[cfg(test)]`? This could introduce a lot of noise for users who are running with `RUST_BACKTRACE=1` for other reasons. e.g. I got this error running an example main function that tries to create a client with a bad URI with `RUST_BACKTRACE=1 cargo run`:

<img width="1208" alt="Screenshot 2024-08-16 at 12 10 23" src="https://github.com/user-attachments/assets/3aaa1338-875d-4714-a779-8e1ae2875dcc">

Reply:
Good call, updated.